compiler/qsc_parse/src/item/tests.rs

```diff
@@ -1557,3 +1557,139 @@ fn disallow_qubit_scoped_block() {
+                            lo: 30,
+                            hi: 39,
+                        },
+                    ),
+                ),
+                Error(
+                    Token(
+                        Eof,
+                        Keyword(
+                            Newtype,
+                        ),
+                        Span {
+                            lo: 116,
+                            hi: 123,
+                        },
+                    ),
+                ),
+            ]"#]],
+    );
+}
+
+#[test]
+fn reject_namespace_with_multiple_nested_levels() {
+    check_vec(
+        parse_namespaces,
+        "namespace LevelOne {
+            namespace LevelTwo {
+                namespace LevelThree {
+                    function DeepFunction() : Unit {}
+                }
+            }
+        }",
+        &expect![[r#"
+            Namespace _id_ [0-146] (Ident _id_ [10-18] "LevelOne"):
+
+            [
+                Error(
+                    Token(
+                        Close(
+                            Brace,
+                        ),
+                        Keyword(
+                            Namespace,
+                        ),
+                        Span {
+                            lo: 33,
+                            hi: 42,
+                        },
+                    ),
+                ),
+                Error(
+                    Token(
+                        Eof,
+                        Close(
+                            Brace,
+                        ),
+                        Span {
+                            lo: 163,
+                            hi: 164,
+                        },
+                    ),
+                ),
+            ]"#]],
+    );
+}
+
+#[test]
+fn namespace_with_attributes_and_docs() {
+    check_vec(
+        parse_namespaces,
+        "/// Documentation for LevelOne
+        namespace LevelOne {
+            @ExampleAttr()
+            /// Documentation for InnerItem
+            function InnerItem() : Unit {}
+        }",
+        &expect![[r#"
+            Namespace _id_ [0-183] (Ident _id_ [49-57] "LevelOne"):
+                doc:
+                    Documentation for LevelOne
+                Item _id_ [72-173]:
+                    Attr _id_ [72-86] (Ident _id_ [73-84] "ExampleAttr"):
+                        Expr _id_ [84-86]: Unit
+                    Callable _id_ [99-173] (Function):
+                        name: Ident _id_ [152-161] "InnerItem"
+                        input: Pat _id_ [161-163]: Unit
+                        output: Type _id_ [166-170]: Path: Path _id_ [166-170] (Ident _id_ [166-170] "Unit")
+                        body: Block: Block _id_ [171-173]: <empty>"#]],
+    );
+}
+
+#[test]
+fn namespace_with_conflicting_names() {
+    check_vec(
+        parse_namespaces,
+        "namespace Conflicts {
+            function Item() : Unit {}
+            newtype Item = Int;
+        }",
```

Comment:
I would've expected this to produce a compilation error as well... should the test verify that in addition to ensuring recovery knows how to produce HIR?

Reply:
Good clarification. Never mind then!